src/main/java/net/redborder/cep/sinks/kafka/KafkaSink.java

```diff
@@ -86,22 +100,25 @@ public void process(String streamName, String topic, Map<String, Object> message
 
     @Override
     public void shutdown() {
+        closed.set(true);
+        producer.flush();
         producer.close();
     }
 
     /**
      * This method sends a given message, with a given key to a given kafka topic.
      *
-     * @param topic The topic where the message will be sent
-     * @param key The key of the message
+     * @param topic   The topic where the message will be sent
+     * @param key     The key of the message
      * @param message The message to send
      */
 
     public void send(String topic, String key, Map<String, Object> message) {
         try {
             String messageStr = objectMapper.writeValueAsString(message);
-            KeyedMessage<String, String> keyedMessage = new KeyedMessage<>(topic, key, messageStr);
-            producer.send(keyedMessage);
+            ProducerRecord<String, String> record = new ProducerRecord<>(topic, key, messageStr);
+            producer.send(record);
+            producer.flush();
```

Comment:
You don't need to do a flush each send. You need to do a flush when shutdown the system.